### ISSUE TITLE
[Feature Request] [CLI] Able to scope down the connection.provider setting to flow folder or better each CLI invoke 

### ISSUE THREAD

**Issue by ttthree** [microsoft/promptflow#753]
The connection.provider setting is a global one, stops the developers from having just a subset of flows using Azure connections.
The main scenario blocked is the Azure AI SDK/CLI integration:

1) I initialized a flow with Azure AI CLI, which set to use Azure connections 
2) Some of my other flows are still using local connections
3) One of #1 and #2 cannot work properly

Possible options:
a) Honor flowfolder/.promptflow/config.yaml for connection setting. This might cause some confusion for users looking at the connections tab in VSC extension.
b) Add a --config connection.provider=x argument to pf CLI (and SDK as well?)

**Comment by danielsc** [comment]
We already hade some discussions around this (use of config.json, etc.). Let's please spec the behavior here and review it to make sure all learnings from the past are considered.

**Comment by brynn-code** [comment]
For CLI: We will expose a suppress argument for cli command, includes `pf test` and `pf run create`, in this case, only full workspace resource id is accepted as argument value, for example:
`pf test  --connection-provider="azureml:/subscriptions/../resourceGroups/../"`
For SDK: We will support connection_provider in kwargs when initializing PFClient, same to cli, only full id is accepted, example usage like:
`PFClient(connection_provider="azureml:/...")`

**Comment by lebaro-msft** [comment]
Why can't the same simple approach of using "azureml" work as it does with pf config --set connection.provider=azureml? The CLI isn't currently keeping track of the full "workspace resource id" or "project resource id" where the connections are. 

I'm not sure why the asymmetry would be wanted by customers or the pf team. Can you clarify why this is a superior design to the previously agreed to approach? 

TL;DR: **we should be able to specify just azureml**

@robch

**Comment by brynn-code** [comment]
@lebaro-msft please take a look at the pr description, see if it meets your requirements, thanks.

**Comment by lebaro-msft** [comment]
That looks great @brynn-code thank you! Let's discuss post-Ignite if there is a more robust solution we can have for this - it sounded like there was a per-flow connection design in discussion we may be interested in.

**Comment by brynn-code** [comment]
To align with our `pf config` command, I made some changes to the interface, please take a look again, thanks! @lebaro-msft

**Comment by brynn-code** [comment]
The issue is marked as closed as I merged the pr, the change would be brought in next 0b9 version release at about 10/25. If you need the dev version for test, please ping me in teams.

**Comment by brynn-code** [comment]
please try with:  pip install promptflow[azure]==0.0.108132407 --extra-index-url https://azuremlsdktestpypi.azureedge.net/promptflow/
